Types/HttpHeader.php

```diff
@@ -41,9 +41,16 @@ public static function create(array $options, array $tokens) {
         $headers = empty($headers) ? [] : $headers;
         $headers = is_string($headers) ? explode(',', $headers) : $headers;
 
+        // Do not send pagination headers if pagination is sent as query params
+        if(!isset($options['pagination_as_query']) || !$options['pagination_as_query']) {
```

Comment:
This one is dangerous. Try to never use branching as statements, but as expressions - in your case as assignment.

Means:
```php
$headers = array_merge(
              $headers,
              !isset($options['pagination_as_query']) || !$options['pagination_as_query'] ?
 PaginationHeaders::create($tokens) : [],
              OrderingHeaders::create($tokens)
          );
```